src/thumbnailer.cc

```diff
@@ -205,13 +217,24 @@ Thumbnailer::Status Thumbnailer::GenerateAnimationEqualPSNR(
           frame_max_quality = frame_mid_quality - 1;
         }
       }
-      config_.quality = frame_final_quality;
 
-      if (!WebPAnimEncoderAdd(enc_, &frame.pic, frame.timestamp_ms, &config_)) {
+      frame.config.quality = frame_final_quality;
+
+      WebPPicture new_pic;
+      WebPPictureInit(&new_pic);
+      if (!WebPPictureCopy(&frame.pic, &new_pic)) {
+        WebPPictureFree(&new_pic);
+        return kMemoryError;
+      }
+
+      if (!WebPAnimEncoderAdd(enc_, &new_pic, frame.timestamp_ms,
+                              &frame.config)) {
         return kMemoryError;
       }
 
-      std::cerr << config_.quality << ' ';
+      WebPPictureFree(&new_pic);
+
+      std::cerr << frame.config.quality << ' ';
```

Comment:
At the moment, the printed `frame.config.quality` is for debugging. I suggest that we save a list of final quality for each frame, which will be used by our plotting tools. Thoughts? @y-guyon @miwucs

Reply:
It seems reasonable, it would be stored as a `float quality;` in `FrameData`? Do you plan to output this list through standard output and/or save it to a file?

Reply:
Is it ok if I store final quality for each frame as `int final_quality` in FrameData ?

Reply:
Yes